### PR TITLE
Distinguish between example and swagger, more specific errror message

### PR DESCRIPTION
**Pull request by ruowan**
Fix [issue22](https://github.com/Azure/avocado/issues/22)

### CODE REVIEW

**Comment by ruowan** [comment]
#22

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Changelog
 
+## 0.5.1
+
+- Distinguish between example and swagger, and ignore analyzing '\$ref' in example file.
+- More specific error message about UNREFERENCED_JSON_FILE, see [issus[](https://github.com/Azure/avocado/issues/22)
```
Comment (nschonni) [inline CHANGELOG.md]:
```suggestion
- More specific error message about UNREFERENCED_JSON_FILE, see [issue 22](https://github.com/Azure/avocado/issues/22)
```

File: src/errors.ts [hunk 2]
```diff
@@ -1,5 +1,16 @@
...
+
 import * as jsonParser from '@ts-common/json-parser'
 
+type ErrorMessage =
+  | 'The example JSON file is not referenced from the swagger file.'
+  | 'The swagger JSON file is not referenced from the readme file.'
+  | 'The `readme.md` is not AutoRest markdown file.'
+  | 'The JSON file is not found but it is referenced from the readme file.'
+  | 'The JSON file exist circular reference.'
+  | 'The file is not valid JSON file.'
```
Comment (nschonni) [inline src/errors.ts]:
```suggestion
  | 'The file is not a valid JSON file.'
```

File: src/errors.ts [hunk 3]
```diff
@@ -1,5 +1,16 @@
...
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
 import * as jsonParser from '@ts-common/json-parser'
 
+type ErrorMessage =
+  | 'The example JSON file is not referenced from the swagger file.'
+  | 'The swagger JSON file is not referenced from the readme file.'
+  | 'The `readme.md` is not AutoRest markdown file.'
+  | 'The JSON file is not found but it is referenced from the readme file.'
+  | 'The JSON file exist circular reference.'
```
Comment (nschonni) [inline src/errors.ts]:
```suggestion
  | 'The JSON file has a circular reference.'
```

File: src/errors.ts [hunk 4]
```diff
@@ -1,5 +1,16 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
 import * as jsonParser from '@ts-common/json-parser'
 
+type ErrorMessage =
+  | 'The example JSON file is not referenced from the swagger file.'
+  | 'The swagger JSON file is not referenced from the readme file.'
+  | 'The `readme.md` is not AutoRest markdown file.'
```
Comment (nschonni) [inline src/errors.ts]:
```suggestion
  | 'The `readme.md` is not an AutoRest markdown file.'
```

File: src/test/avocado-test.ts [hunk 5]
```diff
@@ -173,7 +174,7 @@ describe('avocado', () => {
     const expected = [
       {
         code: 'CIRCULAR_REFERENCE',
-        message: 'The JSON exist circular reference',
+        message: 'The JSON file exist circular reference.',
```
Comment (nschonni) [inline src/test/avocado-test.ts]:
```suggestion
        message: 'The JSON file has a circular reference.',
```

File: src/index.ts [hunk 6]
```diff
@@ -141,6 +146,9 @@ const moveTo = (a: Set<string>, b: Set<string>, key: string): string => {
   return key
 }
 
+const isExample = (filePath: string): boolean =>
+  filePath.split('/').some(name => name === 'examples') || filePath.split('\\').some(name => name === 'examples')
```
Comment (nschonni) [inline src/index.ts]: could use https://nodejs.org/api/path.html#path_path_sep instead

Comment (ruowan) [reply src/index.ts]: Thank you. It's a more elegant way.

File: src/index.ts [hunk 7]
```diff
@@ -120,6 +120,11 @@ type Specification = {
    * readme referenced
    */
   readonly readMePath: string
+
+  /**
+   * kind
+   */
+  readonly kind: 'EXAMPLE' | 'SWAGGER'
```
Comment (raych1) [inline src/index.ts]: It would be better to use enum.

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,10 @@
 # Changelog
 
+## 0.5.1
+
+- Distinguish between example and swagger, and ignore analyzing '\$ref' in example file.
```
Comment (raych1) [inline CHANGELOG.md]: Should we flag this when example file has '$ref' referenced?

File: src/index.ts (hunk 7 above)
Comment (sergey-shandar) [reply src/index.ts]: No, In TypeScript, please use union. Try to avoid enums.

File: CHANGELOG.md (hunk 8 above)
Comment (sergey-shandar) [reply CHANGELOG.md]: No. It's ok to have `$ref` in an example.

**Pull request merged**